client/Components/Overlay.tsx

```diff
@@ -25,4 +29,23 @@ export class Overlay extends React.Component<OverlayProps, OverlayState> {
             {this.props.children}
         </div>;
     }
+
+    public componentDidMount() {
+        this.updateHeight();
+    }
+
+    public componentDidUpdate() {
+        this.updateHeight();
+    }
+
+    private updateHeight() {
+        let domElement = ReactDOM.findDOMNode(this);
+        if (domElement instanceof Element) {
+            let newHeight = domElement.getBoundingClientRect().height;
+            if (newHeight != this.height) {
+                this.height = newHeight;
+                this.forceUpdate();
```

Comment:
It seems like `height` would be more appropriate as part of the component's state so you need not call forceUpdate. Why do it this way?

Reply:
I did it that way simply out of ignorance regarding React. I just pushed a commit to use State instead.